cmd/common-main.go

```diff
@@ -461,25 +461,31 @@ func handleCommonArgs(ctxt serverCtxt) {
 	certsDir := ctxt.CertsDir
 	certsSet := ctxt.certsDirSet
 
-	if consoleAddr == "" {
-		p, err := xnet.GetFreePort()
-		if err != nil {
-			logger.FatalIf(err, "Unable to get free port for Console UI on the host")
-		}
-		// hold the port
-		l, err := net.Listen("TCP", fmt.Sprintf(":%s", p.String()))
-		if err == nil {
-			defer l.Close()
-		}
-		consoleAddr = net.JoinHostPort("", p.String())
+	browserEnabled, e := config.ParseBool(env.Get(config.EnvBrowser, config.EnableOn))
```

Comment:
there is a similar code here  in cmd/server-main.go

``` 
        globalBrowserEnabled, err = config.ParseBool(env.Get(config.EnvBrowser, config.EnableOn))
        if err != nil {
                logger.Fatal(config.ErrInvalidBrowserValue(err), "Invalid MINIO_BROWSER value in environment variable")
        }   
```

I think it is better to move that code to new separate function called serverHandleEarlyEnvVars() and then call  serverHandleEarlyEnvVars()  before serverHandleCmdArgs() in server-main.go, then use globalBrowserEnabled flag in the rest of this commit.